### PR TITLE
feat: rapid label extraction

### PR DESCRIPTION
**Pull request by william-silversmith**
The main connected components algo is reasonably fast. Extracting images however can be far slower. For an image that takes 0.85 seconds to scan, it takes 3+ minutes to extract all the labels using the more obvious numpy operators.

You can do a lot with [`scipy.ndimage.find_objects`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.find_objects.html) and then working with clippings, but there are going to be use cases where you don't want to crop the volume (such as if each voxel needs to align with other data).

Here we work on removing that problem by providing an iterator to each component's isolated image. The iterator is faster than the numpy operation because it can (a) work in-place on a single output image (b) draws only voxels that need to be drawn reducing duplicate work.

```python
for label, img in tqdm(cc3d.each(out, binary=False, in_place=True)):
  process(label, img)
```

In my tests, using the typical numpy operation below took over 3 minutes:

```python
lbls = fastremap.unique(out)
for lbl in tqdm(lbls):
  if lbl == 0:
    continue
  (out == lbl)
```

By contrast, using the `cc3d.series` function took over 1 minute for binary=True, in_place=False, 6 seconds for binary=False,in_place=False, and 3 seconds for in_place=True. Numpy must be treating zeros differently for np.bool.

In other words, the benefit is between 3x to 60x.

### CODE REVIEW

**Comment by william-silversmith** [comment]
This works really well for connectomics images, maybe less so for binary images.
